src/OrleansRuntime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -183,6 +181,8 @@ internal class PersistentStreamPullingAgent : SystemTarget, IPersistentStreamPul
                     "Failed to unregister myself as stream producer to some streams taht used to be in my responsibility.", exc);
             }
             pubSubCache.Clear();
+            IntValueStatistic.Delete(new StatisticName(StatisticNames.STREAMS_PERSISTENT_STREAM_PUBSUB_CACHE_SIZE, StatisticUniquePostfix));          
```

Comment:
Also added Delete to `STREAMS_PERSISTENT_STREAM_PUBSUB_CACHE_SIZE`, mostly just for keeping things clean. There won't be any major memory leak anyway, since we clear the pubSubCache above, but the pubSubCache itself would still be captured, and anything it captures. So better to Delete.